dbms/src/Flash/Pipeline/Exec/PipelineExec.cpp

```diff
@@ -151,44 +153,44 @@ OperatorStatus PipelineExec::executeIO()
 }
 OperatorStatus PipelineExec::executeIOImpl()
 {
-    assert(io_op && *io_op);
-    auto op_status = (*io_op)->executeIO();
-    if (op_status != OperatorStatus::IO)
-        io_op.reset();
-    return op_status;
+    assert(io_op);
+    auto op_status = io_op->executeIO();
+    switch (op_status)
+    {
+    case OperatorStatus::IO:
+        return OperatorStatus::IO;
+    case OperatorStatus::WAITING:
+        fillAwaitable(io_op);
```

Comment:
avoid pass through case here?

Reply:
Do you mean it is better to write `io_op = nullptr; return op_status;` twice?

Reply:
Maybe
```
if (status == STATUS::WAITING)
    fillAwaitable(io_op);
if (status != STATUS::IO)
    io_op = nullptr;
return op_status;
```
?
The difference is that there will be a redundant judgment when the status is waiting.

Reply:
Yeah, just don't prefer fallthrough switch case

Reply:
ok, updated to
```
    assert(io_op);
    auto op_status = io_op->executeIO();
    if (op_status == OperatorStatus::WAITING)
        fillAwaitable(io_op);
    if (op_status != OperatorStatus::IO)
        io_op = nullptr;
    return op_status;
```